Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/System.hs

```diff
@@ -0,0 +1,168 @@
+      setQuery = makeSetClause
+  if Text.null correctWhereClauseText
+    then Nothing
+    else Just $ "UPDATE " <> tableName <> " SET " <> setQuery <> " WHERE " <> correctWhereClauseText <> ";"
+  where
+    makeSetClause :: Text
+    makeSetClause = do
+      Text.intercalate " , " $ map (\(key, val') -> key <> " = " <> valueToSQL val') setClause
+    makeWhereCluase :: Text
+    makeWhereCluase = do
+      Text.intercalate " AND " $ map (\(key, val') -> key <> " = " <> valueToSQL val') whereClause
+
+getKeyNameAndValue :: (MonadThrow m, Log m) => Value -> m [(Text, Value)]
+getKeyNameAndValue (Object obj) = return $ DTX.first DAK.toText <$> DAKM.toList obj
+getKeyNameAndValue _ = throwError $ InvalidRequest "This should have been an object"
+
+queryTypeCheck :: (MonadThrow m, Log m) => Text -> [(Text, Value)] -> [(Text, Value)] -> m ()
+queryTypeCheck tableName' whereClause' setClause' = do
+  let tableName = KP.last $ Text.splitOn "." tableName'
+  mapM_
+    ( \(tableColumn, value) -> do
+        res <- bool (checkParseField tableName tableColumn value) (return True) (tableName' == "atlas_app.system_configs")
+        if res then return True else throwError $ InvalidRequest ("Type Validation Failed For the column: " <> tableColumn)
+    )
+    whereClause'
+
+  mapM_
+    ( \(tableColumn, value) -> do
+        res <- checkParseField tableName tableColumn value
+        if res then return True else throwError $ InvalidRequest ("Type Validation Failed For the column: " <> tableColumn)
+    )
+    setClause'
+
+snakeToCamel :: Text -> Text
+snakeToCamel input = do
+  let parts = Text.splitOn "_" input
+   in case parts of
+        [] -> ""
+        (x : xs) -> Text.pack $ Text.unpack x ++ concatMap capitalize xs
+  where
+    capitalize txt =
+      case Text.unpack txt of
+        [] -> []
+        (y : ys) -> DC.toUpper y : ys
+
+checkParseField :: (MonadThrow m, Log m) => Text -> Text -> Value -> m Bool
+checkParseField tableName tableColumn' value = do
+  let tableColumn = snakeToCamel tableColumn'
+  case tableName of
+    "hot_spot_config" -> return $ checkParse (Proxy @HSC.HotSpotConfigT) tableColumn value
+    "issue_config" -> return $ checkParse (Proxy @ISU.IssueT) tableColumn value
+    "merchant_config" -> return $ checkParse (Proxy @MC.MerchantConfigT) tableColumn value
+    "merchant_service_config" -> return $ checkParse (Proxy @MSC.MerchantServiceConfigT) tableColumn value
+    "merchant_service_usage_config" -> return $ checkParse (Proxy @MSUC.MerchantServiceUsageConfigT) tableColumn value
+    "payout_config" -> return $ checkParse (Proxy @PC.PayoutConfigT) tableColumn value
+    "place_based_service_config" -> return $ checkParse (Proxy @PBSC.PlaceBasedServiceConfigT) tableColumn value
+    "rider_config" -> return $ checkParse (Proxy @RC.RiderConfigT) tableColumn value
+    "ride_related_notification_config" -> return $ checkParse (Proxy @RRNC.RideRelatedNotificationConfigT) tableColumn value
```

Comment:
@manujohnson1234 Can you try generating this with template haskell (so it can support any table)?